integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleUberJarTest.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.gradle;
+
+import org.junit.jupiter.api.Test;
+import org.apache.commons.io.FileUtils;
+import static org.awaitility.Awaitility.await;
+import static io.quarkus.gradle.LaunchUtils.launch;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import io.quarkus.test.devmode.util.DevModeTestUtils;
+import org.awaitility.core.ConditionTimeoutException;
+import static org.assertj.core.api.Assertions.assertThat;
+import static io.quarkus.gradle.LaunchUtils.dumpFileContentOnFailure;
+
+public class MultiModuleUberJarTest extends QuarkusGradleWrapperTestBase {
+    private static Future<?> jarRun;
+
+    @Test
+    public void testUberJarForMultiModule() throws Exception {
+
+        final File projectDir = getProjectDir("uber-jar-for-multi-module-project");
+        runGradleWrapper(projectDir, ":application:quarkusBuild");
+
+        final Path applicationJar = projectDir.toPath().resolve("application").resolve("build");
+        assertThat(applicationJar).exists();
+        Path jar =applicationJar.resolve("application-unspecified-runner.jar");
+        assertThat(jar).exists();
+
+        File output = new File(projectDir, "application/build/output.log");
+        output.createNewFile();
+        Process process = launch(jar, output);
+        try {
+            // Wait until server up
+            dumpFileContentOnFailure(() -> {
+                await()
+                        .pollDelay(1, TimeUnit.SECONDS)
+                        .atMost(1, TimeUnit.MINUTES).until(() -> DevModeTestUtils.getHttpResponse("/hello", 200));
+                return null;
+            }, output, ConditionTimeoutException.class);
+
+            String logs = FileUtils.readFileToString(output, "UTF-8");
+
+            assertThatOutputWorksCorrectly(logs);
+
+            // test that the http response is correct 
+            assertThat(DevModeTestUtils.getHttpResponse("/hello", () -> {
+                return jarRun == null ? null : jarRun.isDone() ? "jar run mode has terminated" : null;
+            }).equals("hello"));
```

Comment:
If this is the actual response, then something is wrong...

```
@GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello " + common.getName();
    }
```

so the response should be `hello common`

Reply:
Oh the response was ```hello common``` only on localhost/8080 , I'll check this again

Reply:
Did you now run the test and make sure it passes?

Reply:
Yes I did, and weird thing is I checked this earlier too, and it was passing then as well, even though the response on http://0.0.0.0:8080/hello was ```hello common``` both the times

Reply:
OK, I'll run it later on and make sure everything is OK

Reply:
Should I mark it ready for review ?

Reply:
Sure